django/urls/resolvers.py

```diff
@@ -329,33 +392,17 @@ def __repr__(self):
             urlconf_repr = repr(self.urlconf_name)
         return '<%s %s (%s:%s) %s>' % (
             self.__class__.__name__, urlconf_repr, self.app_name,
-            self.namespace, self.regex.pattern,
+            self.namespace, self.pattern,
         )
 
     def check(self):
-        warnings = self._check_include_trailing_dollar()
+        warnings = []
         for pattern in self.url_patterns:
             warnings.extend(check_resolver(pattern))
         if not warnings:
-            warnings = self._check_pattern_startswith_slash()
+            warnings = self.pattern.check(None)
         return warnings
 
-    def _check_include_trailing_dollar(self):
-        """
-        Check that include is not used with a regex ending with a dollar.
-        """
-        regex_pattern = self.regex.pattern
-        if regex_pattern.endswith('$') and not regex_pattern.endswith(r'\$'):
-            warning = Warning(
-                "Your URL pattern {} uses include with a regex ending with a '$'. "
-                "Remove the dollar from the regex to avoid problems including "
-                "URLs.".format(self.describe()),
-                id="urls.W001",
-            )
-            return [warning]
-        else:
-            return []
-
     def _populate(self):
         # Short-circuit if called recursively in this thread to prevent
```

Comment:
Seriously wondering why one would even want to support cyclicly nested patterns... From what I can see, the cyclic nature complicates `_populate` (albeit just a bit), but also prevents improvements to the logic of `reverse`. And only using `args` (not kwargs) it brings a possible benefit.